docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -0,0 +1,452 @@
+name that contains the assembly name and optionally a subcomponent name. If an assembly is adding instrumentation for code in a second, independent assembly, the name
+should be based on the assembly that defines the Meter, not the assembly whose code is being instrumented.
+
+- The <xref:System.Diagnostics.Metrics.Meter> constructor version parameter is optional. We recommend that you provide a version in case you release multiple versions
+of the library and make changes to the instruments.
+
+- .NET doesn't enforce any naming scheme for metrics, but by convention all the .NET runtime libraries have metric names using '-' if a separator is needed. Other metric
+ecosystems have encouraged using '.' or '_' as the separator. Microsoft's suggestion is to use '-' in code and let the metric consumer such as OpenTelemetry or
+Prometheus convert to an alternate separator if needed.
+
+#### Viewing the new metric
+
+There are many options to store and view metrics. This tutorial uses the [dotnet-counters](./dotnet-counters.md) tool which is useful for ad-hoc analysis. You can also see
+the [metrics collection tutorial](metrics-collection.md) for other alternatives. If the [dotnet-counters](./dotnet-counters.md) tool is not already installed use the SDK
+to install it:
+
+```dotnetcli
+> dotnet tool update -g dotnet-counters
+You can invoke the tool using the following command: dotnet-counters
+Tool 'dotnet-counters' (version '5.0.251802') was successfully installed.
+```
+
+While the sample app above is still running, list the running processes in a 2nd shell to determine the process ID:
+
+```dotnetcli
+> dotnet-counters ps
+     10180 dotnet     C:\Program Files\dotnet\dotnet.exe
+     19964 metric-instr E:\temp\metric-instr\bin\Debug\netcoreapp3.1\metric-instr.exe
+```
+
+Find the ID for the process name that matches the sample app and have dotnet-counters monitor the new counter:
+
+```dotnetcli
+> dotnet-counters monitor -p 19964 HatCo.HatStore
+Press p to pause, r to resume, q to quit.
+    Status: Running
+
+[HatCo.HatStore]
+    hats-sold (Count / 1 sec)                          4
+```
+
+As expected we can see that HatCo store is steadily selling 4 hats each second. 
+
+### Types of Instruments
+
+Above the example created a <xref:System.Diagnostics.Metrics.Counter> instrument but there are several different types available. Instruments differ in two ways:
+
+- **Default metric computations** - Tools that collect and analyze the instrument measurements will compute different default metrics depending on the instrument.
+- **Storage of aggregated data** - Most useful metrics need data to be aggregated from many measurements. One option is the caller provides individual measurements
+  at arbitrary times and the collection tool will manage the aggregation. Alternately the caller can manage the aggregate measurements and provide them on-demand in a callback.
+
+Types of instruments currently available:
```

Comment:
would it be good to give some examples when each instrument can be used. For example, Counter can be used for counting number of requests. ObservableGauge can be used to report room temperature,...etc.

Reply:
There is a section lower in the text that is 'Best practices selecting an instrument type' which does have this info. Do you think I should restructure to bring the examples closer, repeat the examples, or something else?

Reply:
@tarekgh - making sure you see the question here for you : )

Reply:
I suggest bringing the section `Best practices selecting an instrument type` closer to this section. As the article reader, will give immediate idea why we have such instruments and when to use each one. I am ok if you feel otherwise too. this is just a suggestion :-)

Reply:
FYI - we have some guidance regarding which instrument to be used in the OpenTelemetry spec repo https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#instrument-selection.